lib/api/apiUtils/bucket/bucketCreation.js

```diff
@@ -165,7 +165,8 @@ function createBucket(authInfo, bucketName, headers,
     const ownerDisplayName =
         authInfo.getAccountDisplayName();
     const creationDate = new Date().toJSON();
-    const objectLockEnabled = headers['x-amz-bucket-object-lock-enabled'];
+    const objectLockEnabled
+        = headers['x-amz-bucket-object-lock-enabled'] === 'true';
```

Comment:
I thought headers was a string>string map so it would always be a string.
Also should you add a toLower here in case of "TRUE" or "True"?

Reply:
What about "on", "yes", "1" and 1?

Reply:
I would only include what amazon is also expecting, so if they support any sort of truthy string such as on, yes, etc then I agree. If it explicitly only supports true/false then I wouldn't include anymore.

Reply:
Yes, indeed it needs to be first checked against aws to see what it supports.

Reply:
@miniscruff that's a good catch; seems like both 'true' or 'True' are covered by AWS, @nicolas2bert those are not supported by AWS.

Reply:
For string comparisons, I always recommend case insensitive or locale comparisons.

Reply:
Yeah, I can see how it makes sense; i.e. when I digged deeper I realized that when the request made via CLI with just the presence of the `--object-lock-enabled-for-bucket` flag without any value (this is how CLI command works), when looked into how it works under the hood, seems like it converts it's value as `x-amz-bucket-object-lock-enabled: True` whereas a request via sdk it only allows boolean and converts it lowercase as `x-amz-bucket-object-lock-enabled: True`